code/Player/Controllers/HumanController.cs

```diff
@@ -75,7 +91,7 @@ public override void Simulate()
                 else
                 {
                     ApplyFriction(GroundSurface.Friction * SurfaceFriction);
-                    Walk();
+                    Move();
```

Comment:
That is not only move, this is when we're moving on ground. Maybe call this function MoveGround then ? Just "Move" is too general

Reply:
yeah but the method moves you and don't care if youre on ground

Reply:
It applies the logic of ground, air move is not the same, swim move would not be the same and also no gravity live.
Depending on the state of the player, different logic is applied to make him move

Reply:
Done do you prefer ?